### PR TITLE
AI Remote Control Fixes

### PR DESCRIPTION
**Pull request by Ben10083**
Fixes https://github.com/Aurorastation/Aurora.3/issues/13267
Fixes https://github.com/Aurorastation/Aurora.3/issues/16881
Fixes https://github.com/Aurorastation/Aurora.3/issues/14039
Fixes https://github.com/Aurorastation/Aurora.3/issues/14062

Also adds QoL where pressing eject on remote controlled mechs exits remote control.

### CODE REVIEW

**Comment by Ben10083** [comment]
!bugfix

**Comment by Ben10083** [comment]
WIP until I can fix spiderbot radios

**Comment by Ben10083** [comment]
!wip

**Comment by Ben10083** [comment]
!review

File: code/controllers/subsystems/virtual_reality.dm [hunk 1]
```diff
@@ -230,7 +249,7 @@ SUBSYSTEM_DEF(virtualreality)
 
 	mind_transfer(user, robot[choice])
 
-/datum/controller/subsystem/virtualreality/proc/bound_selection(var/user, var/network)
+/datum/controller/subsystem/virtualreality/proc/bound_selection(var/user, var/network, var/return_list)
```
Comment (FluffyGhoster) [inline code/controllers/subsystems/virtual_reality.dm:252]: `return_list` seems useless here, you can always return the list as it doesn't seem you return anything else

File: code/controllers/subsystems/virtual_reality.dm [hunk 2]
```diff
@@ -173,7 +188,7 @@ SUBSYSTEM_DEF(virtualreality)
 	if(null_vr_mob)
 		target.vr_mob = null
 
-/datum/controller/subsystem/virtualreality/proc/mech_selection(var/user, var/network)
+/datum/controller/subsystem/virtualreality/proc/mech_selection(var/user, var/network, var/return_list)
```
Comment (FluffyGhoster) [inline code/controllers/subsystems/virtual_reality.dm:191]: `return_list` seems useless here, you can always return the list as it doesn't seem you return anything else

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

**Pull request merged**